src/Logging.js

```diff
@@ -0,0 +1,380 @@
+/**
+ * Client Side Logging module.
+ * @module Logging
+ */
+
+// Note that we have an excessive amount of try/catch blocks in this code.
+// That is intentional. Logging should never break the features which are using it
+
+var TRACE_LEVEL_SILENT = 0;
+var TRACE_LEVEL_ALWAYS = 1;
+var TRACE_LEVEL_ERROR = 2;
+var TRACE_LEVEL_WARN = 3;
+var TRACE_LEVEL_INFO = 4;
+var TRACE_LEVEL_LOG = 5;
+var TRACE_LEVEL_DEBUG = 6;
+var TRACE_LEVEL_VERBOSE = 6;
+
+var LOCAL_STORAGE_KEY_NAME = 'AppNexus_Page_Debug_Log_Level';
```

Comment:
seems to be lots of AppNexus references sprinkled throughout the code. We should rename all of these.

Reply:
ok